src/ext_manifest.c

```diff
@@ -81,6 +82,87 @@ static int ext_man_validate(uint32_t section_size, const void *section_data)
 	}
 }
 
+/*
+ * Insert `section_name` content after `elem_type` header.
+ * Save result in new buffer, then replace `meta_section_data`.
+ * Return `section_name` size or error code.
+ */
+static int ext_man_update_element_with_list(const struct module *module,
+					    int32_t *metadata_size,
+					    uint8_t **metadata_buff,
+					    enum ext_man_elem_type elem_type,
+					    const char *section_name)
+{
+	struct ext_man_elem_header* head = NULL;
+	uint8_t *meta_buff = *metadata_buff;
```

Comment:
Can you confirm we are converting a UUID to a memory address (as stated in commit message) ? This is a security hole, we cannot pass in addresses from outside of FW, we can however pass in data we can validate like an index in an array. @lbetlej @mmaka1 any comments here ?

Reply:
@lgirdwood The code is not merged yet to kernel, but my understanding what is passed is a key to the dictionary, not an actual address. @keyonjie @ktrzcinx can you clarify?

Reply:
Yes, it is only a index-liked "key", not the real memory address, those real 16 Bytes UUIDs are not stored in DSP SRAM actually. @mmaka1 please correct me if I am wrong.

Reply:
This is address of UUID entry, from invalid memory region (0x1FFFA000 is already used as start address, but it's fully configurable in linker script) - so dereference leads to firmware crash. This address is used as UUID element identifier/representation from FW side because full UUID value is inaccessible from there. So this address is UUID dictionary key value.

Reply:
ok, can you update the commit messages, inline comments and PR description to use key/index rather than address.

Reply:
done